app/src/main/java/com/example/househomey/filter/ui/TagFilterFragment.java

```diff
@@ -1,22 +1,145 @@
+ * functionality based on tags using {@link TagFilter}.
+ */
 package com.example.househomey.filter.ui;
 
+import android.content.Context;
 import android.view.View;
-
+import androidx.core.content.ContextCompat;
+import com.example.househomey.Item;
+import com.example.househomey.R;
 import com.example.househomey.filter.model.FilterCallback;
+import com.example.househomey.filter.model.TagFilter;
+import com.google.android.material.chip.Chip;
+import com.google.android.material.chip.ChipGroup;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
 
+/**
+ * This class represents a fragment for tag-based filtering. It extends {@link FilterFragment}
+ * and implements methods to manage and interact with tag filters.
+ * @author Jared Drueco
+ */
 public class TagFilterFragment extends FilterFragment {
+    protected TagFilter tagFilter;
+    private final ChipGroup tagChipGroup;
+    private Map<String, Boolean> tagSelectionMap = new HashMap<>();
+
+    /**
+     * Constructs a new {@code TagFilterFragment} with the given parameters.
+     *
+     * @param title           The title of the fragment.
+     * @param contentView     The content view of the fragment.
+     * @param filterCallback  The callback for filter actions.
+     * @param itemList        The list of items to be filtered.
+     */
+    public TagFilterFragment(String title, View contentView, FilterCallback filterCallback, ArrayList<Item> itemList) {
+        super(title, contentView, filterCallback);
+        tagChipGroup = contentView.findViewById(R.id.chip_group_labels);
+        populateTagChipGroup(itemList);
+    }
 
-    public TagFilterFragment(String title, View contentView, FilterCallback filterCallback) {
+    /**
+     * Constructs a new {@code TagFilterFragment} with the given parameters, including a
+     * pre-existing {@link TagFilter}.
+     *
+     * @param title           The title of the fragment.
+     * @param contentView     The content view of the fragment.
+     * @param filterCallback  The callback for filter actions.
+     * @param itemList        The list of items to be filtered.
+     * @param tagFilter       The pre-existing tag filter.
+     */
+    public TagFilterFragment(String title, View contentView, FilterCallback filterCallback, ArrayList<Item> itemList, TagFilter tagFilter) {
         super(title, contentView, filterCallback);
+        this.tagFilter = tagFilter;
+        tagSelectionMap = tagFilter.tagSelectionMap;
+        tagChipGroup = contentView.findViewById(R.id.chip_group_labels);
+        tagChipGroup.removeAllViews();
+        populateTagChipGroup(itemList);
+    }
+
+    /**
+     * Populates the tag {@link ChipGroup} with tags from the provided item list.
+     *
+     * @param itemList The list of items containing tags.
+     */
+    private void populateTagChipGroup(ArrayList<Item> itemList) {
```

Comment:
So here is my concern with this method of storing tags. A tag lives and dies with it's last item. I think we should clarify with the TA if that is the expectation. Take these two examples

We make the "shoe" tag, but don't apply it to any items. Should the tag be an option the next time you open the dialog?

We delete the last shoe, but then we add a new shoe. Should we be able to reuse that tag or do we have to make a new one.

Also do we want tags to be deletable? It doesn't seem to be explicitly stated in the requirements. If we already have answers to these question lmk, otherwise we should ask the TA before we go any further as that's gonna impact a bunch of this

CC@owencooke  @MatthewNeufeld

Reply:
@ldbonkowski I agree that the technical implementation of tags, those aspects you mentioned, need to be discussed in more detail before we merge these two tag branches.

We could set up a team meeting before Tuesday, pick what we think is the best strategy, and confirm with the TA?

Reply:
@ldbonkowski 100% agree we need to get on the same page or else it's just gonna be a mess trying to get everything aligned. I don't think deleting tags is a requirement but it will be super easy to implement and I think having that functionality would be wanted by a user and makes sense to have.

I like @owencooke's meeting idea. Let's do that. You raise some good points @ldbonkowski.